homeassistant/components/mazda/sensor.py

```diff
@@ -1,187 +1,176 @@
 """Platform for Mazda sensor integration."""
-from homeassistant.components.sensor import SensorEntity
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import (
+    CONF_UNIT_SYSTEM_IMPERIAL,
     LENGTH_KILOMETERS,
     LENGTH_MILES,
     PERCENTAGE,
     PRESSURE_PSI,
 )
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+from homeassistant.util.unit_system import UnitSystem
 
 from . import MazdaEntity
 from .const import DATA_CLIENT, DATA_COORDINATOR, DOMAIN
 
 
-async def async_setup_entry(hass, config_entry, async_add_entities):
+@dataclass
+class MazdaSensorEntityDescription(SensorEntityDescription):
+    """Describes a Mazda sensor entity."""
+
+    # Suffix to be appended to the vehicle name to obtain the sensor name
+    name_suffix: str | None = None
+
+    # Function to determine whether the vehicle supports this sensor, given the coordinator data
+    is_supported: Callable[[dict], bool] = lambda data: True
+
+    # Function to determine the unit of measurement for this sensor, given the configured unit system
+    unit: Callable[[UnitSystem], str | None] = lambda unit_system: None
+
+    # Function to determine the value for this sensor, given the coordinator data and the configured unit system
+    value: Callable[
+        [dict, UnitSystem], StateType | None
+    ] = lambda data, unit_system: None
+
+
+SENSOR_ENTITIES = [
+    MazdaSensorEntityDescription(
+        key="fuel_remaining_percentage",
+        name_suffix="Fuel Remaining Percentage",
+        icon="mdi:gas-station",
+        unit=lambda unit_system: PERCENTAGE,
+        is_supported=lambda data: data["status"]["fuelRemainingPercent"] is not None,
+        value=lambda data, unit_system: data["status"]["fuelRemainingPercent"],
+    ),
+    MazdaSensorEntityDescription(
+        key="fuel_distance_remaining",
+        name_suffix="Fuel Distance Remaining",
+        icon="mdi:gas-station",
+        unit=lambda unit_system: LENGTH_MILES
```

Comment:
If the lambda spans more than one line we can define a normal function and assign that function to the attribute.

Reply:
OK, that will be a lot of functions, but I can do that.

Reply:
You can see an example in `renault\sensor.py`

Reply:
@MartinHjelmare I replaced the multi-line lambdas with functions defined at the module level.